lib/primitiv.gi

```diff
@@ -215,6 +215,18 @@ local dom,deg,PD,s,cand,a,p,s_quot,b,cs,n,beta,alpha,i,ag,bg,q,gl,hom;
     fi;
   fi;
 
+  # AbelianInvariants
+  if Length(cand)>1 then
+    a:= AbelianInvariants(grp);
+    b:= [];
+    for i in [1..Length(cand)] do
+      b[i]:= AbelianInvariants(p[i]);
+    od;
+    s:= Filtered([1..Length(cand)], i->b[i] =a);
```

Comment:
Why not just this, and then get rid of the line which init `b` above?
```suggestion
    s:= Filtered([1..Length(cand)], i-> AbelianInvariants(p[i]) =a);
```
Anyway, of course it is also fine this way, just wondering :-)

Reply:
I was duplicating the structure of the surrounding tests (in the hope that reduces accidental errors and makes the test not stick out)